docs/tap_hold.md

```diff
@@ -198,6 +198,38 @@ bool get_retro_tapping(uint16_t keycode, keyrecord_t *record) {
 }
 ```
 
+## Retro Shift
+
+Holding and releasing a Tap Hold key without pressing another key will result in only the hold. With Retro Shift enabled this action will also produce a shifted version of the tap keycode on release.
+
+This is a supplement to [Auto Shift](feature_auto_shift.md), which does not support Tap Hold.  Auto Shift will be enabled automatically if it is not already, but you should read the Auto Shift documentation and configure the Auto Shift timeout matching the tapping term.  All additional Auto Shift defines are respected, including keyrepeat.
+
+Retro Shift does not require [Retro Tapping](#retro-tapping) to be enabled, and if both are enabled the state of Retro Tapping will only apply if the tap keycode is not matched by Auto Shift. `RETRO_TAPPING_PER_KEY`, however, is checked before Retro Shift if defined.
+
+To enable Retro Shift, add the following to your `config.h`:
+
+```c
+#define RETRO_SHIFT
```

Comment:
If this does use functions from `process_auto_shift`, then this should be a make/rules thing, not a define.

Reply:
Why? Permissive hold, ignore mod tap interrupt, and even the 'stock' retro tapping are all `#defines`. This is just retro tapping for Auto Shift, and only in its own `#define` because it makes the hold have *two* possible actions. This is not an alternative to Auto Shift's normal functionality.

Reply:
> If this does use functions from `process_auto_shift`

If it's actually using functions from the C file, then it needs to be sourced, too.  And if that's the case, the way to do that is via the make system.  

If it's just using defines from it, then ... I'm not sure they belong in the auto shift header file, then, and should probably go into the action_tapping.h file.

Reply:
It is *inside* `process_auto_shift`. As above, like retro tapping, this is a modifier for the (Auto Shift's) behavior with Tap Holds that are released without using their hold action on another key. Any references to the `#define` outside `process_auto_shift` (such as in `action.c` and `action_tapping.c`, as the hold action needs to be prevented sometimes because it cannot be captured and interpreted easily) also check if Auto Shift is enabled.

Reply:
Ah, okay. In that case, I would prefer that the documentation stay in the auto-shift doc, personally.   That way, it's less prone to confusion.   Maybe with a subnote about this functionality in autoshift.